external/fv3fit/fv3fit/data/netcdf/load.py

```diff
@@ -0,0 +1,157 @@
+import contextlib
+import logging
+from dataclasses import dataclass
+from typing import Callable, Iterator, Mapping, Optional, Sequence
+
+import numpy as np
+import tensorflow as tf
+import xarray as xr
+from fv3fit.data.base import TFDatasetLoader, register_tfdataset_loader
+from fv3fit.tfdataset import iterable_to_tfdataset
+from toolz.functoolz import compose_left
+
+from .io import CACHE_DIR, download_cached, get_nc_files
+
+logger = logging.getLogger(__name__)
+
+__all__ = ["NCDirLoader", "nc_dir_to_tfdataset"]
+
+
+@contextlib.contextmanager
+def open_dataset(path: str) -> Iterator[xr.Dataset]:
+    ds = xr.open_dataset(path)
+    yield ds
+    ds.close()
```

Comment:
```suggestion
    ds = xr.open_dataset(path)
    try:
        yield ds
    finally:
        ds.close()
```

Reply:
Thanks...I decided to remove the context manager and use try-finally directly: cf25cb8c3bbd34d6d131f997090928192b02f14c